language-guides/spring-data-neo4j/spring-data-neo4j.adoc

```diff
@@ -34,129 +34,206 @@ When developing with Neo4j, please use JDK 8 and your favorite IDE.
+There is a Person node and a Movie node, so a domain class for each is needed.
+In each, the class is annotated with @NodeEntity to map that these objects will be nodes in the graph.
+Each contains an ID field that is annotated as the id and populated with a value generated by the database (@GeneratedValue).
+After the id, additional fields are set up to hold different information we want to capture about the object.
+
+The last few lines of each class map the relationship between the nodes, which are connected by the Role that the Person played in the Movie.
+Both the Person and Movie classes are annotated with @Relationship and name the type of relationship as "ACTED_IN".
+The direction property is outgoing by default, so we must specify that the relationship is incoming on the Movie node.
+To link Person and Movie together, a new entity called Role must be created.
+In this class, we annotate it as a relationship entity and specify the name of the relationship (ACTED_IN).
+The Role class also has an ID field that is generated by the database and an array list to contain a possible list of roles (person could play more than one role in a movie).
+Finally, annotations are added to mark the Person and Movie nodes as start and end nodes for the relationship.
+We now have our graph structure mapped in our application. This is the object-graph mapping (OGM) piece.
 
 .Entities
 [source,java]
 ----
 @NodeEntity
 public class Movie {
 
-  @Id @GeneratedValue Long id;
+    @Id @GeneratedValue private Long id;
+	private String title;
+	private int released;
+	private String tagline;
 
-  String title;
+	@JsonIgnoreProperties("movie")
+	@Relationship(type = "ACTED_IN", direction = Relationship.INCOMING)
+	private List<Role> roles;
+}
+
+@NodeEntity
+public class Person {
 
-  Person director;
+    @Id @GeneratedValue private Long id;
+	private String name;
+	private int born;
 
-  @Relationship(type="ACTED_IN", direction = Relationship.INCOMING)
-  Set<Person> actors = new HashSet<>();
+	@Relationship(type = "ACTED_IN")
+	private List<Movie> movies = new ArrayList<>();
 }
+
+@RelationshipEntity(type = "ACTED_IN")
+public class Role {
+
+    @Id @GeneratedValue private Long id;
+	private List<String> roles = new ArrayList<>();
+
+	@StartNode
+	private Person person;
+
+	@EndNode
+	private Movie movie;
 ----
 
+The repository interface allows the developer to create methods and queries needed for retrieving the data from the database.
+The interface extends the Neo4jRepository, which extends the Spring CRUDRepository for persisting and retrieving data.
+Some basic queries are derived for us (like the findByTitleLike method), but others must be specifically written and annotated with @Query.
+
 .Declare a repository interface
 [source,java]
 ----
-interface MovieRepository extends Neo4jRepository<Movie, Long> {
+public interface MovieRepository extends Neo4jRepository<Movie, Long> {
+
+    @Query("MATCH (m:Movie {title: {title}}) RETURN m LIMIT 1")
```

Comment:
`@Query` is not needed here, probably use a different example, like findByReleased or findByActorsName?